### ISSUE TITLE
Feature request: Include currently processed template to config.build

### ISSUE THREAD

**Issue by jsbrain** [maizzle/framework#754]
I want to determine the file/template currently being processed in the events hook, like `beforeRender`.

What I want to achieve is copy over an asset into the final build directory that is dynamically decided based on a frontmatter prop of the current template.

So let's say I have `templates/template-A/index.html` and `templates/template-B/index.html` and I want to have a certain asset copied right into the root folder of the resulting build directories.

What I want to do is:

```ts
async beforeRender(html, config) {
    if (config?.site?.logo) {
      // Resolve the logo from the local filesystem
      const logoDir = path.resolve(__dirname, "src", "assets", "logos");

      const logoFileName = fs
        .readdirSync(logoDir)
        .find((file) => file.startsWith(`logo-${config.site.logo}`));

      if (logoFileName) {
        const logoPath = path.join(logoDir, logoFileName);
        const destination = path.resolve(
          __dirname,
          "..",
          config.build.templates.destination.path,
          config.build.currentTemplateName // <- I'm missing a way to determine the current template
          //                                     name so I can copy it to the right build directory
        );

        // copy over to build folder
        fs.copyFileSync(logoPath, path.join(destination, logoFileName));
      }
    }

    // must always return the `html`
    return html;
  },
```

So the resulting build directory would look like:

```
dist
├── template-A
│   ├── index.html
│   └── logo-A.png
└── template-B
    ├── index.html
    └── logo-B.png
```

I know this is an extreme edge case but I think adding a property that has the path to the current template could be beneficial for rare cases (like mine) and I think it should be easy to add without introducing any breaking changes!?

**Comment by cossssmin** [comment]
I think we had it at some point but it was undocumented and must have gotten lost in transition to 4.0.

I'll add it back, it'll be a parsed path of the template that is currently being compiled, i.e.:

```js
build: {
    templates: {
      // ...
    },
    current: {
      root: '',
      dir: 'build_production',
      base: 'transactional.html',
      ext: '.html',
      name: 'transactional'
    }
  }
```

Since Maizzle copies the source files to destination and then compiles them there, the parsed path is your `build.templates.destination.path`, hence the `build_production` directory.

Also, this'll only be available when using the CLI with `maizzle build` - there's no source file path when using the API.

**Comment by cossssmin** [comment]
Have a feeling we might need to add more stuff to `current`, so I think I'll release it under `build.current.path` instead, so we don't end up introducing breaking changes if we want to add more stuff in there.

```js
current: {
  path: {
    {
      root: '',
      dir: 'build_production',
      base: 'transactional.html',
      ext: '.html',
      name: 'transactional'
    }
  }
}
```

**Comment by cossssmin** [comment]
Released in 4.1.0, feel free to reopen if anything 👍

https://github.com/maizzle/framework/releases/tag/v4.1.0

**Comment by jsbrain** [comment]
Amazing! Thank you 😇
